src/waveform/renderers/waveformmark.cpp

```diff
@@ -127,14 +136,242 @@ bool WaveformMark::contains(QPoint point, Qt::Orientation orientation) const {
     // Without some padding, the user would only have a single pixel width that
     // would count as hovering over the WaveformMark.
     float lineHoverPadding = 5.0;
-    int position;
-    if (orientation == Qt::Horizontal) {
-        position = point.x();
-    } else {
-        position = point.y();
+    if (orientation == Qt::Vertical) {
+        point = QPoint(point.y(), m_breadth - point.x());
     }
-    bool lineHovered = m_linePosition >= position - lineHoverPadding &&
-            m_linePosition <= position + lineHoverPadding;
+    bool lineHovered = m_linePosition >= point.x() - lineHoverPadding &&
+            m_linePosition <= point.x() + lineHoverPadding;
 
     return m_label.area().contains(point) || lineHovered;
 }
+
+// Helper class to calculate the geometry and fontsize needed by generateImage
+// to draw the label and text
+struct MarkerGeometry {
+    bool m_isSymbol; // it the label normal text or a single symbol (e.g. open circle arrow)
+    QFont m_font;
+    QRectF m_contentRect;
+    QRectF m_labelRect;
+    QSizeF m_imageSize;
+
+    MarkerGeometry(const QString& label, bool useIcon, Qt::Alignment align, float breadth) {
+        // If the label is 1 character long, and this character isn't a letter or a number,
+        // we can assume it's a special symbol
+        m_isSymbol = label.length() == 1 && !label[0].isLetterOrNumber();
+
+        // This alone would pick the OS default font, or that set by Qt5 Settings (qt5ct)
+        // respectively. This would mostly not be notable since contemporary OS and distros
+        // use a proven sans-serif anyway. Though, some user fonts may be lacking glyphs
+        // we use for the intro/outro markers for example.
+        // So, let's just use Open Sans which is used by all official skins to achieve
+        // a consistent skin design.
+        m_font.setFamily("Open Sans");
+        // For text, use a pixel size like everywhere else in Mixxx, which can be scaled
+        // well in general.
+        // Point sizes would work if only explicit Qt scaling QT_SCALE_FACTORS is used,
+        // though as soon as other OS-based font and app scaling mechanics join the
+        // party the resulting font size is hard to predict (affects all supported OS).
+
+        m_font.setPixelSize(13);
+        m_font.setWeight(75); // bold
+        m_font.setItalic(false);
+
+        const qreal margin{3.f};
+
+        qreal capHeight;
+        {
```

Comment:
why is this scoping needed?

Reply:
oh is this copied from the original code? then maybe that's fine 👍🏻

Reply:
i removed it, as it allows capHeight to be const.